crates/hir-def/Cargo.toml

```diff
@@ -33,11 +33,15 @@ base-db = { path = "../base-db", version = "0.0.0" }
 syntax = { path = "../syntax", version = "0.0.0" }
 profile = { path = "../profile", version = "0.0.0" }
 hir-expand = { path = "../hir-expand", version = "0.0.0" }
+rustc_abi = { version = "0.0.20221111", package = "hkalbasi-rustc-ap-rustc_abi", default-features = false }
 mbe = { path = "../mbe", version = "0.0.0" }
 cfg = { path = "../cfg", version = "0.0.0" }
 tt = { path = "../tt", version = "0.0.0" }
 limit = { path = "../limit", version = "0.0.0" }
 
+[target.wasm32-unknown-unknown.dependencies]
+getrandom = { version = "0.2.8", features = ["js"] }
```

Comment:
Where does the requirement for randomness come from?

Reply:
`rustc_abi` uses random for `-Z randomize-layout` I think. We also can simulate `-Z randomize-layout` by feeding `field_shuffle_seed` in `ReprOptions`.

Reply:
But it might be a better idea to disable `getrandom` feature of `rand` as we don't use the seed from the os?

Reply:
> But it might be a better idea to disable getrandom feature of rand as we don't use the seed from the os?

Yeah

Reply:
I disabled the feature on the rustc side. Will remove this when I bumped the dependency.

Reply:
Do we even need to support randomized layouts at all? That seems pretty much useless to us here, now I don't know how integrated that is into the rustc code but cfg'ing `rand` out would be pretty nice as we don't really need the dependency.